### PR TITLE
Dockerfile / requirements.txt: Remove unused dependencies

### PR DESCRIPTION
**Pull request by AurelienGasser**
Follow-up to https://github.com/SubstraFoundation/substra-backend/pull/216

### CODE REVIEW

**Comment by AurelienGasser** [comment]
@Kelvin-M @inalgnu I tried something more... blunt. Let me know what you think!

Tested with end-to-end tests: :heavy_check_mark: 

---

Stats for the `substra-backend` docker image: (should be similar for flower, worker, etc):

**Before**:
- build time: [04:02](https://console.cloud.google.com/cloud-build/builds/a800bab2-c188-4ac8-9382-f5915b94a2ab;step=1;tab=detail?project=substra-208412)
- image size: [638.2MB](https://console.cloud.google.com/gcr/images/substra-208412/GLOBAL/substra-backend@sha256:ca01b26f70642ff7388cc25ef8704905c5235b442c4c4997f5bfb1a0f1c16102/details?tab=info&project=substra-208412)

**After**:
- build time: [03:47](https://console.cloud.google.com/cloud-build/builds/993dd19d-91f9-4a46-9d21-802f17429a6f;step=1;tab=detail?project=substra-208412)
- image size: [543.8MB](https://console.cloud.google.com/gcr/images/substra-208412/GLOBAL/substra-backend@sha256:1d4e903501a7cd11cb2a0799273541edcfea665ed963d41db689f78d109ae91c/details?tab=info&project=substra-208412)

**Comment by AurelienGasser** [comment]
The failing unit test is `substrapp.tests.tests_exception.ExceptionTests`, which is flaky/nondeterministic AFAIK (but I could be missing something)

File: docker/celeryworker/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 FROM nvidia/cuda:9.2-base-ubuntu18.04
 
-RUN apt-get update
-RUN apt-get install -y python3.6 python3-pip python3-dev build-essential libssl-dev libffi-dev libxml2-dev libxslt1-dev zlib1g-dev
-RUN apt-get install -y git curl netcat
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends python3-dev python3-pip python3-setuptools gcc git && \
```
Comment (AlexandrePicosson) [inline docker/celeryworker/Dockerfile:4]: As you are not installing recommended package anymore `python3-wheel` is not installed anymore breaking the docker build (at least locally).

Removing `build-essentials` also seems to break the build as you need `gcc` and `g++` to build the python packages as they are not pre-built for the cuda image as they are for an ubuntu image for example.

Comment (AurelienGasser) [reply docker/celeryworker/Dockerfile]: @AlexandrePicosson Good catch ! I must have made a mistake when I tested.

**Comment by AurelienGasser** [comment]
phew

**Comment by AurelienGasser** [comment]
This is ready for another review.

**Pull request merged**